src/server/StudioManagementSystem.Infrastructure/Repositories/ProjectRepository.cs

```diff
@@ -47,10 +48,33 @@ public async Task<Guid> AddProjectAsync(Project project, CancellationToken ct)
             throw new DataException($"Cannot create a {nameof(Project)} with an existing name, '{project.Title}");
         }
 
-        await _smsDbContext.Projects.AddAsync(project, ct);
-        await _smsDbContext.SaveChangesAsync(ct);
+        Guid returnId;
 
-        return project.Id;
+        try {
+            var principalOwner = (await _ownerContactRepository.GetOwnersByIdAsync(
+                new List<Guid> { project.PrincipalOwnerId }, ct)
+            ).FirstOrDefault();
+
+            if (principalOwner is null) {
+                throw new DataException($"Cannot assign a non-existent ${nameof(OwnerContact)} with id: '${project.PrincipalOwnerId}' as the principal owner to a ${nameof(Project)}");
+            }
+
+            project.PrincipalOwner = principalOwner;
+
+            await _smsDbContext.Projects.AddAsync(project, ct);
+            await _smsDbContext.SaveChangesAsync(ct);
+            returnId = project.Id;
+        } catch (Exception ex) {
+            _logger.LogError(ex, "An exception occured while assigning the principal owner to a new {Project}", nameof(Project));
+
+            if (ex is DataException) {
+                ExceptionDispatchInfo.Capture(ex).Throw();
```

Comment:
we definitely want to throw the DataException on failing to link the principal owner contact, others we can log and move on as they're not known edge cases